pkg/cache/cache.go

```diff
@@ -986,3 +989,30 @@ func (c *Cache) MatchingClusterQueues(nsLabels map[string]string) sets.Set[kueue
 func queueKey(q *kueue.LocalQueue) queue.LocalQueueReference {
 	return queue.NewLocalQueueReference(q.Namespace, kueue.LocalQueueName(q.Name))
 }
+
+// AddOrUpdateDynamicResourceAllocationConfig ingests the singleton CRD and updates
+// the in-memory DeviceClass → logical resource map.
+// It is idempotent and thread-safe.
+func (c *Cache) AddOrUpdateDynamicResourceAllocationConfig(log logr.Logger, draCfg *kueuealpha.DynamicResourceAllocationConfig) {
+	c.Lock()
+	defer c.Unlock()
+
+	// Rebuild from scratch to avoid stale entries.
+	newMap := make(map[corev1.ResourceName]corev1.ResourceName)
+	for _, res := range draCfg.Spec.Resources {
+		for _, dc := range res.DeviceClassNames {
+			newMap[dc] = res.Name
+		}
+	}
+	c.draDeviceClassToResource = newMap
+	log.V(2).Info("Updated DRA DeviceClass mapping", "entries", len(newMap))
+}
+
+// GetResourceNameForDeviceClass converts a DeviceClass to its logical resource name.
+// Returns (logicalRes, true) on success or ("", false) when unmapped.
+func (c *Cache) GetResourceNameForDeviceClass(dc corev1.ResourceName) (corev1.ResourceName, bool) {
+	c.RLock()
+	defer c.RUnlock()
+	lr, ok := c.draDeviceClassToResource[dc]
+	return lr, ok
```

Comment:
```suggestion
	return c.draDeviceClassToResource[dc]
```

Simplify the implementations.

Reply:
Oh, that makes sense.
Thank you for checking that.